src/export.rs

```diff
@@ -63,10 +67,34 @@ pub fn unexport(pin_config: &PinConfig,
 ///
 /// If the GPIO is already exported, this function will continue
 /// without an error as the desired end state is achieved.
-pub fn export(pin_config: &PinConfig, symlink_root: Option<&str>) -> Result<(), sysfs_gpio::Error> {
+pub fn export(pin_config: &PinConfig, symlink_root: Option<&str>) -> Result<()> {
     let pin = pin_config.get_pin();
     try!(pin.export());
 
+    // change user, group, mode for files in gpio directory
+    for entry in fs::read_dir(format!("/sys/class/gpio/gpio{}", &pin_config.num))? {
+        let e = entry?;
+        let metadata = e.metadata()?;
+
+        let user = pin_config.user.as_ref().and_then(|username| get_user_by_name(username));
+        let group = pin_config.group.as_ref().and_then(|groupname| get_group_by_name(groupname));
+
+        if metadata.is_file() {
+            if user.is_some() && group.is_some() {
```

Comment:
If the user specified in the config file is invalid, I think this will just silently not set the permissions. That's not a great experience.